x/iro/keeper/invariants.go

```diff
@@ -0,0 +1,70 @@
+package keeper
+
+import (
+	"errors"
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dymensionxyz/dymension/v3/utils/uinv"
+	"github.com/dymensionxyz/dymension/v3/x/iro/types"
+)
+
+var invs = uinv.NamedFuncsList[Keeper]{
+	{Name: "plan", Func: InvariantPlan},
+}
+
+func RegisterInvariants(ir sdk.InvariantRegistry, k Keeper) {
+	invs.RegisterInvariants(types.ModuleName, ir, k)
+}
+
+// DO NOT DELETE
+func AllInvariants(k Keeper) sdk.Invariant {
+	return invs.All(types.ModuleName, k)
+}
+
+// the plan should validate and bookkeeping for tokens should be sensible
+func InvariantPlan(k Keeper) uinv.Func {
+	return uinv.AnyErrorIsBreaking(func(ctx sdk.Context) error {
+		plans := k.GetAllPlans(ctx, false)
+		if len(plans) == 0 {
+			return nil
+		}
+
+		var errs []error
+		for _, plan := range plans {
+			err := checkPlan(plan)
+			err = errorsmod.Wrapf(err, "planID: %d", plan.Id)
+			errs = append(errs, err)
```

Comment:
why u need first `append` then `errors.Join`? can't u `errors.Join` them directly?

Reply:
Hmm I guess you could, it's not really the intended use of join, you'd have recursive lists
Resolving iydm